pkg/alexa/gen/skill_test.go

```diff
@@ -217,25 +226,27 @@ func TestSkillBuilder_Build_Full(t *testing.T) {
 	assert.Error(t, err)
 }
 
+// Cover skill validations
 func TestSkillRestrictions(t *testing.T) {
 	sb := gen.NewSkillBuilder()
 
-	// no category
+	// no locales
 	_, err := sb.Build()
 	assert.Error(t, err)
 
-	// no testing instructions (missing locales)
-	sb.WithCategory(alexa.CategoryDating)
+	// no category
+	en := sb.AddLocale("en-US")
```

Comment:
split your test cases.

Reply:
I refactored `skill_test.go`, please have a look if that's better.
I also refactored some functions to make them work more consistently, will do that for `model.go` as well:
```go
SkillBuilder().WithModel() // will create and *set* a single ModelBuilder and return the *SkillBuilder*
SkillBuilder().AddLocale("en-US") // will create and *add* a single LocaleBuilder and return the *SkillBuilder*
SkillBuilder().Locale("en-US") // will return the named *LocaleBuilder*
```

so all `With` and `Add` functions return the builder itself so you can chain nicely. The functions with only the name (`Locale()`) let you "switch" to the sub-object in the chain, if you want.